lib/utils/compose_ts.ts

```diff
@@ -775,7 +889,22 @@ async function newTarDirectory(
 				mode: fileStats.stats.mode,
 				size: fileStats.stats.size,
 			},
-			await readFile(fileStats.filePath),
+			fileStats.relPath === configPath
+				? setConfig(configPath, config, parsedBuildArgs)
+				: await readFile(fileStats.filePath),
+		);
+	}
+	if (
+		configPath === '' &&
+		(!_.isEmpty(parsedBuildArgs.global) || !_.isEmpty(parsedBuildArgs.services))
```

Comment:
I believe this logic is correct -- if configPath is empty, then the for loop `for (fileStats of filteredFileList)` ought to have found a matching file and called `setConfig`. But I had to stop and think about it for a moment. Would it be clearer to have a `foundConfig` boolean variable that was set to true just before calling `setConfig` if it is ever called? Then here having the test be:
```
	if (
		!foundConfig &&
		(!_.isEmpty(parsedBuildArgs.global) || !_.isEmpty(parsedBuildArgs.services))
```